### ISSUE TITLE
Decode to typed fields fails

### ISSUE THREAD

**Issue by themihai** [gorilla/schema#31]
Decoder (actually parsePath) doesn't recognise the field names if they are "typed" or if they and/or if they are embedded. See the tests below.

internal/testdata

	package testdata
	type Aa bool

decoder_test.go

	import "./internal/testdata"
	s1 := &struct{
		 testdata.Aa
	}{}
	v1 := map[string][]string{
		"Aa": {"true"},
	}
	decoder := NewDecoder()
	_ = decoder.Decode(s1, v1)
	if s1.Aa != testdata.Aa(true) {
		t.Errorf("s1: expected %v, got %v", true, s1.Aa)
	}

	v2 := map[string][]string{
		"F01": {"true"},
	}
	s2 := &struct{
		F01 testdata.Aa
	}{}
	_ = decoder.Decode(s2, v2)
	if s2.F01 != testdata.Aa(true) {
		t.Errorf("s2: expected %v, got %v", true, s2.F01)
	}

**Comment by kisielk** [comment]
Hm yes, that should work but I guess there is a bug there.

**Comment by kisielk** [comment]
I think this is going to take an overhaul of the decoding system to fix. The converters need to work based on the underlying kind and then using the appropriate `SetT` method in the reflect package. `encoding/json`'s `decoder.go` is a good example of how it should work.

**Comment by bfrog** [comment]
ran in to this same exact issue myself when trying to decode into something like...

```go
type EnumType int
type EnumTypes []EnumTypes

type DecodeIntoStruct struct {
   Types EnumTypes `schema:"types"`
}
```

In this case it fails on the reflect.Append call

**Comment by bfrog** [comment]
This looks like its fixed with the text unmarshaler PR

**Comment by kisielk** [comment]
Do you mean #30?

**Comment by themihai** [comment]
It seems this was fixed on #39 so I'm closing it for now.
